tests/testsuite/git.rs

```diff
@@ -3332,28 +3332,28 @@ fn metadata_master_consistency() {
                 }
               ],
               "workspace_members": [
-                "foo 0.1.0 [..]"
+                "[..]foo#0.1.0"
               ],
               "workspace_default_members": [
-                "foo 0.1.0 [..]"
+                "[..]foo#0.1.0"
               ],
               "resolve": {
                 "nodes": [
                   {
-                    "id": "bar 1.0.0 (__BAR_SOURCE__#__BAR_HASH__)",
+                    "id": "file:///[..]/bar#1.0.0",
```

Comment:
I worry that `PackageIdSpec` is ambiguous and will need to be extended before we can move forward with this
- It loses the semantics on top of the raw protocol (PackageId exposed this as `git+file`)
- It tracks the version instead of the hash

Any other problems people can find?

Reply:
If we do make changes, we should keep #7725 in mind

Reply:
We talked about this at the cargo team meeting and were generally ok with moving this forward.